diff -ruN fontconfig-2.10.0/src/Makefile.in _fontconfig-2.10.0/src/Makefile.in
--- fontconfig-2.10.0/src/Makefile.in	2012-07-17 01:18:58 -0400
+++ _fontconfig-2.10.0/src/Makefile.in	2013-02-13 15:06:59 -0500
@@ -436,6 +436,16 @@
 	  echo "rm -f \"$${dir}/so_locations\""; \
 	  rm -f "$${dir}/so_locations"; \
 	done
+
+clean-dll:
+	rm -f *.dll
+	rm -f *.def
+	rm -f *.exp
+	rm -f *.lib
+
+libfontconfig.dll:
+	$(CC) -o libfontconfig-1.dll -s -shared -Wl,--out-implib=.libs/libfontconfig.dll.a -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--whole-archive .libs/*.o -Wl,--no-whole-archive $(libfontconfig_la_LIBADD) $(LIBS)
+
 libfontconfig.la: $(libfontconfig_la_OBJECTS) $(libfontconfig_la_DEPENDENCIES) $(EXTRA_libfontconfig_la_DEPENDENCIES) 
 	$(AM_V_CCLD)$(libfontconfig_la_LINK) -rpath $(libdir) $(libfontconfig_la_OBJECTS) $(libfontconfig_la_LIBADD) $(LIBS)
 
@@ -636,7 +646,7 @@
 @ENABLE_SHARED_FALSE@install-data-local:
 clean: clean-am
 
-clean-am: clean-generic clean-libLTLIBRARIES clean-libtool \
+clean-am: clean-generic clean-libLTLIBRARIES clean-dll clean-libtool \
 	clean-noinstPROGRAMS mostlyclean-am
 
 distclean: distclean-am
@@ -735,13 +745,16 @@
 @OS_WIN32_FALSE@install-libtool-import-lib:
 @OS_WIN32_FALSE@uninstall-libtool-import-lib:
 
-@MS_LIB_AVAILABLE_TRUE@fontconfig.lib : libfontconfig.la
+@MS_LIB_AVAILABLE_TRUE@fontconfig.lib : libfontconfig.la libfontconfig.dll
 @MS_LIB_AVAILABLE_TRUE@	lib -name:libfontconfig-@LIBT_CURRENT_MINUS_AGE@.dll -def:fontconfig.def -out:$@
 
 @MS_LIB_AVAILABLE_TRUE@install-ms-import-lib:
 @MS_LIB_AVAILABLE_TRUE@	$(INSTALL) fontconfig.lib $(DESTDIR)$(libdir)
+@MS_LIB_AVAILABLE_TRUE@	$(INSTALL) libfontconfig-1.dll $(DESTDIR)$(bindir)
+@MS_LIB_AVAILABLE_TRUE@	ln -fs libfontconfig-1.dll $(DESTDIR)$(bindir)/libfontconfig.dll
 
 @MS_LIB_AVAILABLE_TRUE@uninstall-ms-import-lib:
+@MS_LIB_AVAILABLE_TRUE@	$(RM) $(DESTDIR)$(bindir)/libfontconfig.*
 @MS_LIB_AVAILABLE_TRUE@	$(RM) $(DESTDIR)$(libdir)/fontconfig.lib
 
 @MS_LIB_AVAILABLE_FALSE@install-ms-import-lib:
