$ diff -u postgis-2.0.3/configure.ac postgis-2.0.3/configure2.ac
--- postgis-2.0.3/configure.ac  2012-12-20 12:38:24 -0500
+++ postgis-2.0.3/configure2.ac 2013-08-06 16:44:45 -0400
@@ -727,7 +727,7 @@

                dnl Add the include directory to JSON_CPPFLAGS
                JSON_CPPFLAGS="-I$JSONDIR/include"
-               JSON_LDFLAGS="-L$JSONDIR/lib -ljson"
+               JSON_LDFLAGS="-L$JSONDIR/lib -ljson -ljson-c"
        fi
 fi

@@ -746,7 +746,7 @@
 if test "$HAVE_JSON" = "yes"; then
        AC_DEFINE([HAVE_LIBJSON], 1, [Define to 1 if libjson is present])
     if test "x$JSON_LDFLAGS" = "x"; then
-        JSON_LDFLAGS="-ljson"
+       JSON_LDFLAGS="-ljson -ljson-c"
     fi
 fi

