--- readline-6.3/shlib/Makefile.in	2012-01-12 16:54:34 -0500
+++ readline-6.3/shlib/Makefile2.in	2014-08-09 23:13:06 -0400
@@ -46,6 +46,7 @@
 CP = cp
 MV = mv
 LN = ln
+LIBS = @LIBS@
 
 SHELL = @MAKE_SHELL@
 
@@ -86,7 +87,7 @@
 SHOBJ_LIBS = @SHOBJ_LIBS@
 
 SHLIB_XLDFLAGS = @LDFLAGS@ @SHLIB_XLDFLAGS@
-SHLIB_LIBS = @SHLIB_LIBS@
+SHLIB_LIBS = @SHLIB_LIBS@ $(LIBS)
 
 SHLIB_DOT = @SHLIB_DOT@
 SHLIB_LIBPREF = @SHLIB_LIBPREF@
--- readline-6.3/shell.c	2013-03-13 10:17:00 -0400
+++ readline-6.3/shell2.c	2014-08-09 20:25:37 -0400
@@ -4,7 +4,7 @@
 /* Copyright (C) 1997-2009 Free Software Foundation, Inc.
 
    This file is part of the GNU Readline Library (Readline), a library
-   for reading lines of text with interactive input and history editing.      
+   for reading lines of text with interactive input and history editing.
 
    Readline is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -191,7 +191,7 @@
 sh_unset_nodelay_mode (fd)
      int fd;
 {
-#if defined (HAVE_FCNTL)
+#if defined (HAVE_FCNTL) && defined(F_GETFL)
   int flags, bflags;
 
   if ((flags = fcntl (fd, F_GETFL, 0)) < 0)
--- readline-6.3/input.c	2014-01-10 15:07:08 -0500
+++ readline-6.3/input2.c	2014-08-09 20:03:45 -0400
@@ -3,7 +3,7 @@
 /* Copyright (C) 1994-2013 Free Software Foundation, Inc.
 
    This file is part of the GNU Readline Library (Readline), a library
-   for reading lines of text with interactive input and history editing.      
+   for reading lines of text with interactive input and history editing.
 
    Readline is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -206,7 +206,7 @@
     return -1;
 #endif
 
-#if defined (O_NDELAY)
+#if defined (O_NDELAY) && defined(F_GETFL)
   if (result == -1)
     {
       tem = fcntl (tty, F_GETFL, 0);
@@ -344,8 +344,8 @@
 
 void
 _rl_insert_typein (c)
-     int c;     
-{    	
+     int c;
+{
   int key, t, i;
   char *string;
 
@@ -441,7 +441,7 @@
 	    {
 	      if (rl_get_char (&c) != 0)
 		break;
-		
+
 	      if ((r = rl_gather_tyi ()) < 0)	/* XXX - EIO */
 		{
 		  rl_done = 1;
@@ -557,7 +557,7 @@
   memset(&ps, 0, sizeof (mbstate_t));
   memset(&ps_back, 0, sizeof (mbstate_t));
 
-  mb_len = 0;  
+  mb_len = 0;
   while (mb_len < size)
     {
       RL_SETSTATE(RL_STATE_MOREINPUT);
@@ -577,7 +577,7 @@
 	  /* shorted bytes */
 	  ps = ps_back;
 	  continue;
-	} 
+	}
       else if (mbchar_bytes_length == 0)
 	{
 	  mbchar[0] = '\0';	/* null wide character */
--- readline-6.3/configure.ac	2014-02-11 17:12:40 -0500
+++ readline-6.3/configure2.ac	2014-08-10 01:01:05 -0400
@@ -136,7 +136,7 @@
 
 AC_HEADER_STDC
 
-AC_HEADER_STAT
+#AC_HEADER_STAT
 AC_HEADER_DIRENT
 
 AC_CHECK_FUNCS(fcntl kill lstat)
