--- tcl8.6.1/win/configure.in   2013-09-19 16:17:14 -0400
+++ tcl8.6.1/win/configure2.in  2013-10-24 16:14:29 -0400
@@ -294,7 +294,7 @@

 eval "TCL_DLL_FILE=tcl${VER}${DLLSUFFIX}"

-eval "TCL_STUB_LIB_FILE=\"${LIBPREFIX}tclstub${VER}${LIBSUFFIX}\""
+eval "TCL_STUB_LIB_FILE=\"${LIBPREFIX}tclstub${VER}.a\""
 eval "TCL_STUB_LIB_FLAG=\"-ltclstub${VER}${LIBFLAGSUFFIX}\""
 eval "TCL_BUILD_STUB_LIB_SPEC=\"-L`pwd` ${TCL_STUB_LIB_FLAG}\""
 eval "TCL_STUB_LIB_SPEC=\"-L${libdir} ${TCL_STUB_LIB_FLAG}\""

--- tcl8.6.1/win/tcl.m4 2013-09-19 16:17:14 -0400
+++ tcl8.6.1/win/tcl2.m4        2013-10-24 15:18:23 -0400
@@ -693,6 +693,7 @@
            runtime=
            LIBRARIES="\${STATIC_LIBRARIES}"
            EXESUFFIX="s\${DBGX}.exe"
+            LIBSUFFIX="\${DBGX}.a"
        else
            # dynamic
             AC_MSG_RESULT([using shared flags])
@@ -708,6 +709,7 @@

            EXESUFFIX="\${DBGX}.exe"
            LIBRARIES="\${SHARED_LIBRARIES}"
+            LIBSUFFIX="\${DBGX}.dll.a"
        fi
        # Link with gcc since ld does not link to default libs like
        # -luser32 and -lmsvcrt by default. Make sure CFLAGS is
@@ -715,11 +717,10 @@
        SHLIB_LD='${CC} -shared ${CFLAGS}'
        SHLIB_LD_LIBS='${LIBS}'
        MAKE_DLL="\${SHLIB_LD} \$(LDFLAGS) -o \[$]@ ${extra_ldflags} \
-           -Wl,--out-implib,\$(patsubst %.dll,lib%.a,\[$]@)"
+           -Wl,--out-implib,\$(patsubst %.dll,lib%.dll.a,\[$]@)"
        # DLLSUFFIX is separate because it is the building block for
        # users of tclConfig.sh that may build shared or static.
        DLLSUFFIX="\${DBGX}.dll"
-       LIBSUFFIX="\${DBGX}.a"
        LIBFLAGSUFFIX="\${DBGX}"
        SHLIB_SUFFIX=.dll
