--- bdwgc-gc7_4_2/include/private/gc_locks.h	2014-06-03 02:09:52 -0400
+++ bdwgc-gc7_4_2/include/private/gc_locks2.h	2014-08-15 12:30:08 -0400
@@ -87,7 +87,7 @@
      /* integers for each thread, though that should be true as much    */
      /* as possible.                                                    */
      /* Refine to exclude platforms on which pthread_t is struct */
-#    if !defined(GC_WIN32_PTHREADS)
+#    if !defined(GC_WIN32_PTHREADS) || defined(__MINGW32__)
 #      define NUMERIC_THREAD_ID(id) ((unsigned long)(id))
 #      define THREAD_EQUAL(id1, id2) ((id1) == (id2))
 #      define NUMERIC_THREAD_ID_UNIQUE
